src/zenml/pipelines/pipeline_definition.py

```diff
@@ -565,19 +566,26 @@ def build(
             if settings:
                 compile_args["settings"] = settings
 
-            deployment, _, _ = self._compile(**compile_args)
-            pipeline_id = self._register().id
-
-            local_repo = code_repository_utils.find_active_code_repository()
-            code_repository = build_utils.verify_local_repository_context(
-                deployment=deployment, local_repo_context=local_repo
+            _from_config_file = self._parse_config_file(
```

Comment:
Not a huge fan of you having to do this manually before the compilation step, but I also don't see any easy way around this. I'll refactor this (the entire config file logic is very messy right now) after this PR is merged.